lib/ansible/module_utils/tacp_ansible/tacp_utils.py

```diff
@@ -9,10 +9,223 @@
 from tacp.rest import ApiException
 
 from ansible.module_utils.tacp_ansible.tacp_exceptions import ActionTimedOutException, InvalidActionUuidException
+from ansible.module_utils.tacp_ansible.tacp_constants import Action
 from uuid import UUID, uuid4
 from time import sleep
 
 
+class ApplicationResource(object):
+    def __init__(self, api_client):
+        self.api = tacp.ApplicationsApi(api_client)
+        self.api_client = api_client
+
+    def get_all(self):
+        """ Get all application instances as a list of dicts
+        """
+        api_response = self.api.get_applications_using_get()
+        instance_dicts = []
+        if api_response:
+            for instance in api_response:
+                instance_dicts.append(instance.to_dict())
+        return instance_dicts
+
+    def get_uuid_by_name(self, name):
+        """ Return the UUID of an appplication instance if it exists
+            given the name of the instance, None otherwise
+        """
+        all_instances = self.get_all()
+        for instance in all_instances:
+            if instance['name'] == name:
+                return instance['uuid']
+        return None
+
+    def get_by_uuid(self, uuid):
+        """ Returns an application as a dict specified by UUID, None otherwise
+        """
+        api_response = self.api.get_application_using_get(uuid)
+        if api_response:
+            return api_response.to_dict()
+        return None
+
+    def create(self, body):
+        """ Creates an application with the provided API request body, returns
+            bool of success/failure
+        """
+        api_response = self.api.create_application_from_template_using_post(
+            body)
+        try:
+            wait_for_action_to_complete(
+                api_response.action_uuid, self.api_client)
+            return api_response.object_uuid
+        except ActionTimedOutException:
+            return "Exception when waiting for action to complete, action timed out."
+        except InvalidActionUuidException:
+            return "Exception when waiting for action to complete, invalid action UUID."
+        except Exception as e:
+            return e
+
+    def delete(self, uuid):
+        """ Deletes an application specified by UUID, returns bool of success/failure
+        """
+        api_response = self.api.delete_application_using_delete(uuid)
+        try:
```

Comment:
Would recommend taking the code doing the `wait_for_action_to_complete` and putting in a separate function since it is being repeated a lot. Ideally, you could create a base class that all the `Resource` classes inherit from and add this as a private method there.

Reply:
Sounds good. Also a `promise` or some other name decorator would be perfect for making these methods "sync", api wise